webapp/channels/src/components/sidebar/sidebar_channel/sidebar_group_channel/sidebar_group_channel.tsx

```diff
@@ -25,40 +25,43 @@ type Props = {
     };
 };
 
-type State = Record<string, never>;
-
-export default class SidebarGroupChannel extends React.PureComponent<Props, State> {
-    handleLeaveChannel = (callback: () => void) => {
-        const id = this.props.channel.id;
+const SidebarGroupChannel = ({
+    channel,
+    currentUserId,
+    actions,
+    active,
+    currentTeamName,
+    redirectChannel,
+    membersCount,
+}: Props) => {
+    const handleLeaveChannel = useCallback((callback: () => void) => {
+        const id = channel.id;
         const category = Constants.Preferences.CATEGORY_GROUP_CHANNEL_SHOW;
 
-        const currentUserId = this.props.currentUserId;
-        this.props.actions.savePreferences(currentUserId, [{user_id: currentUserId, category, name: id, value: 'false'}]).then(callback);
+        actions.savePreferences(currentUserId, [{user_id: currentUserId, category, name: id, value: 'false'}]).then(callback);
 
         trackEvent('ui', 'ui_direct_channel_x_button_clicked');
 
-        if (this.props.active) {
-            getHistory().push(`/${this.props.currentTeamName}/channels/${this.props.redirectChannel}`);
+        if (active) {
+            getHistory().push(`/${currentTeamName}/channels/${redirectChannel}`);
         }
-    };
+    }, [channel.id, actions, active, currentTeamName, redirectChannel, currentUserId]);
 
-    getIcon = () => {
+    const getIcon = useCallback(() => {
         return (
-            <div className='status status--group'>{this.props.membersCount}</div>
+            <div className='status status--group'>{membersCount}</div>
         );
-    };
+    }, [membersCount]);
```

Comment:
Since we are calling the function in line 62 instead of passing it as a prop, it is safe not to use a useCallback here.